### ISSUE TITLE
sumit --dryrun fails with http error 400 when submission to TW failed

### ISSUE THREAD

**Issue by belforte** [dmwm/CRABServer#4912]
ERROR: RuntimeError: Reading
https://cmsweb.cern.ch/crabcache/logfile?name=dry-run-sandbox.tar.gz
failed with code 400

could a more descriptive message be printed for users ?
possibly including the reason why crab submit failed.

see:
https://hypernews.cern.ch/HyperNews/CMS/get/computing-tools/818.html

**Comment by matz-e** [comment]
@belforte, @mmascher: we can check the status of the last command from within the client and not do the dry-run when the command failed. As seen in the HN message, this would give you a failure mode.  In the message, you can also see that the dry-run continues despite the submission failing.

Drawback: When the splitting fails, the dry-run would not run…

I'm trying to think of some other way to solve this that's not too involved…

**Comment by belforte** [comment]
the cleanest way would be to separate the FAILSUBMIT status into SPLITFAIL and SUBMITFAIL. @mmascher would it make sense to do this via some variable that the client can lookup to tell the two, w/o asking you to change more the new StateMachine ?

**Comment by matz-e** [comment]
As I've mentioned in #4915, it may actually be better to move the dry-run before the splitting, in which case this would be moot. Running the dry-run before then would be good too, IMHO, because I can construct some scenarios where the dry-run does not see enough events to give good estimates.

**Comment by belforte** [comment]
I agree that moving dry-run is better, but we still should separate (if nothing else for our monitoring and to help ops figure out what's going on) failed in splitting vs. failed to talk to schedd.
